src/amber/controller/helpers/responders.cr

```diff
@@ -51,9 +62,11 @@ module Amber::Controller::Helpers
 
     private def requested_responses
       req_responses = Array(String).new
-
-      if (accept = context.request.headers["Accept"]?) && !accept.empty?
-        accepts = accept.split(";").first?.try(&.split(/,|,\s/))
+      path_ext = request.path.match(Content::REGEX).try(&.[1])
+      if (path_ext)
+        req_responses << Content::TYPE[path_ext]
+      elsif (accept = context.request.headers["Accept"]?) && !accept.empty?
+        accepts = accept.split(";").first?.try(&.split(Content::COMMA_OR_COMMASPACE_REGEX))
```

Comment:
This is my inner ruby-ist thinking out loud.

Take it or leave it.

```crystal
private def type_from_path_extension : String | Nil
  return @type_from_path_extension if @type_from_path_extension_checked
  @type_from_path_extension_checked = true
  @type_from_path_extension = request.path.match(Content::REGEX).try(&.[1])
end

private def accepts_from_headers : Array(String)
  @accepts_from_headers ||= parse_accepts_from_headers
end

private def parse_accepts_from_headers : Array(String)
  accept_header = context.request.headers["Accept"]?
  if accept_header
    accept_header.split(";").first?.try(&.split(Content::COMMA_OR_COMMASPACE_REGEX))
  else
    [] of String
  end
end

private def requested_responses : Array(String)
  return [type_from_path_extension] unless type_from_path_extension.nil?
  accepts_from_headers
end
```

Reply:
Is this easier to read to you? I do like the idea of not defining the Array in the beginning.

Reply:
I guess the main thing for me here is avoiding as many branches and memory allocations as I can.

Reply:
Maybe this:
```crystal
private def type_from_path_extension : String | Nil
  request.path.match(Content::REGEX).try(&.[1])
end

private def parse_accepts_from_headers : Array(String)
  accept_header = context.request.headers["Accept"]?
  if accept_header
    accept_header.split(";").first?.try(&.split(Content::COMMA_OR_COMMASPACE_REGEX))
  else
    [] of String
  end
end

private def requested_responses : Array(String)
  if (type_from_path = type_from_path_extension)
    [type_from_path]
  else
    parse_accepts_from_headers
  end
end
```